text/0017-message/0017-message.md

```diff
@@ -0,0 +1,181 @@
+6. If the Message PoW Hash will contain at least the number of trailing 0 trits the node defines as required.
+
+### Payloads
+
+Each message will contain a payload. The specification of the payloads is out of scope of this RFC. Below is a table of the currently specified core payloads with a link to their specifications. The `Unsigned Payload` will be specified here as an example.
+
+| Payload Name                              |   Type Value |
+| ---------------------------------------   | -----------  | 
+|  Signed Transaction                       |     0        |
+|  Signed Data                              |     1        |
+|  [Unsigned Data](#Unsigned-Data-Payload)  |     2        |
+| Indexation                                |     3        |
+
+### Unsigned Data Payload
+
+To make the Payload concept clear we will define the `unsigned data payload`. As the name suggests it simply allows to add arbitrary data to a message. A message that has been attached to the tangle and approved by a milestone has useful properties: You can verify that the content of the data did not change, and you can ascertain the approximate time it was published by checking the approving milestone. For example, one can record a hash of a text document on the tangle with such a payload. When a third party requests a proof that the document was published at a certain time and was not changed, one can point them to them to the message containing the `unsigned data payload` with the hash.
+
+The structure of the payload is simple:
+
+
+
+| Name             | Type     | Description               |
+| --------         | -------- | -----------               |
+| Payload Type     | varint   | Must be set to **2**      |
+| Data             | ByteArray| The data we are attaching |
+
+There are no validation rules for the payload. Message validation rules suffice here.
+
+
+
+# Rationale and alternatives
+
+Instead of creating a layered approach, we could have simply created a flat`transaction message`, that is tailored for mutating the ledger state, and try to fit all the use cases there. For example with the unsigned data use-case, we could have filled some section of the transaction with the data. Then via a flag in the transaction we could have instructed to not pass this transaction to the service that attempts to mutate the ledger state.
+
+This approach seems less extensible. It might have made sense if we wanted to build a protocol that is just for ledger mutating transactions, but we want to be able to extend the protocol to do more than that.
+
+# Unresolved questions
+
+- What should be the maximum length of the message?
+- Better protection with PoW? how does the length of the message and type of payloads should affect difficulty?
+- How should we pad the message to fit it into CURL for PoW?
+- How should we pad the input for CURL? One proposal is to go with padding method 2 that is described here https://en.wikipedia.org/wiki/ISO/IEC_9797-1#Padding
```

Comment:
I was trying to get some response from researchers about this one but maybe I didn't do it hard enough

@Wollac offred the padding rule and I think it is good enough because even if someone manages to speed up PoW it won't be the most troubling thing

Reply:
Has been resolved with #24